jdaviz/tests/test_app.py

```diff
@@ -17,3 +17,58 @@ def test_get_tray_item_from_name():
 
     with pytest.raises(KeyError, match='not found in app'):
         app.get_tray_item_from_name('imviz-compass')
+
+
+def test_duplicate_data_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    dc = specviz_helper.app.data_collection
+    assert dc[0].label == "test"
+    assert dc[1].label == "test (1)"
+    specviz_helper.load_spectrum(spectrum1d, data_label="test_1")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    assert dc[2].label == "test_1"
+    assert dc[3].label == "test (2)"
+
+
+def test_duplicate_data_labels_with_brackets(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    dc = specviz_helper.app.data_collection
+    assert len(dc) == 2
+
+
+def test_return_data_label_is_none(specviz_helper):
+    data_label = specviz_helper.app.return_data_label(None)
+    assert data_label == "Unknown"
+
+
+def test_return_data_label_is_image(specviz_helper):
+    data_label = specviz_helper.app.return_data_label("data/path/test.jpg")
+    assert data_label == "test[jpg]"
+
+
+def test_hdulist_with_filename(cubeviz_helper, image_cube_hdu_obj):
+    image_cube_hdu_obj.file_name = "test"
+    data_label = cubeviz_helper.app.return_data_label(image_cube_hdu_obj)
+    assert data_label == "test[HDU object]"
+
+
+@pytest.mark.filterwarnings('ignore:.* is a low contrast image')
+def test_file_path_not_image(imviz_helper, tmp_path):
+    path = tmp_path / "myimage.fits"
+    path.mkdir()
+    data_label = imviz_helper.app.return_data_label(str(path))
+    assert data_label == "myimage"

```

Comment:
Is this test misleading though? Imviz is supposed to attach extension info to a FITS data.

Reply:
This test directly tests the `return_data_label` method, there are plenty of imviz tests that test adding extensions back onto filenames (I had to fix quite a few for this PR to work).